src/features/Authentication/userSlice.ts

```diff
@@ -48,8 +56,26 @@ export const selectUserId = createSelector(
 );
 
 export const selectUserRole = createSelector(
+  selectUserState,
+  ({ userRole }) => userRole ?? 'mentee',
+);
+
+export const selectAppRole = createSelector(
   selectUserState,
   selectChatsExist,
-  ({ userRole }, hasUserChats) =>
-    userRole === 'mentee' && !hasUserChats ? 'freshMentee' : userRole,
+  ({ userRole }, hasUserChats) => {
+    if (userRole === 'mentee' && !hasUserChats) return 'freshMentee';
+    if (!userRole) return 'mentee';
+    return userRole;
+  },
+);
```

Comment:
Hmm this is already very nitpicky/beautifying-coment, but maybe we could even utilize the selectUserRole selector here:

```typescript
export const selectAppRole = createSelector(
  selectUserRole,
  selectChatsExist,
  ({ userRole }, hasUserChats) => {
    if (userRole === 'mentee' && !hasUserChats) return 'freshMentee';
    return userRole;
  },
);
```

we can remove the `if (!userRole) return 'mentee';`